### ISSUE TITLE
PWM switching freqency and related noise

### ISSUE THREAD

**Issue by dedekcz** [ottoragam/Tarocco#19]
Hi,

Do You know if is possible to increase PWM switching frequency from 6 kHz to 24 kHz (or above 20 kHz) to reduce noise? In my system, I have very low encoder error (less then 3 points) but sometimes motor stay at final position, and it looks that regulation still keeping to get target and this leads to noise.

Do You have such experience?
Thanks,
Ros.

**Comment by ottoragam** [comment]
Hi Ros,
It should be possible to change the PWM frequency by adjusting the clock of the Timer/Counter block in PSoC Creator that feeds the pulses to the gate driver. This will also increase the load on the microcontroller, as the PID output calculation rate is tied to the PWM frequency, but I believe it should be able to handle 20+kHz. You can play with it if you wish, or give me a pair of days so I can test this myself and post the updated firmware.

**Comment by dedekcz** [comment]
Yes, I believe on your expertise, So I will wait for your FW. Many thanks.

**Comment by ottoragam** [comment]
I made available a new firmware version. The pulse freq should be around 23 kHz now. Please test it and let me know your findings.

**Comment by dedekcz** [comment]
Hi,

I tried to build FW same way,  but after update to Driver I lost Serial communication. For confirmation of my process, I repeat build procedure with FW (2017-08-12) again, and serial connection works there.
So it looks that new FW has some issue.

**Comment by ottoragam** [comment]
Sorry for the lack of communication, I've been very busy these past days. I'll check to see what might be wrong with the new firmware and write back.

**Comment by labkab** [comment]
Hi
I have the same issue with audible sound for servomotor
so flashed new ultrasonic pwm frequency firmware but does not work.
Symptom is that motor spins in one direction without command without stopping without error..
With old firmware works fine..

**Comment by petetom** [comment]
Hello,
I have the same issues with the latest firmware 03-21-2021.  I noticed that after programming with this latest version both LEDs do not stop flashing after programed.  Power is cycled LEDs stop flashing after 3 sec as expected and with no pulse inputs the motor moves and not able to create any faults like disconnecting the encoder signals.  Also there is no debug info coming back from the serial port.

Have you been able to get the this version to properly work?
The only way to program is to use the no-optimization GCC-493 path for the dependency. 

Thanks,
-pete
